agent/engine/dockerstate/docker_task_engine_state.go

```diff
@@ -494,34 +494,20 @@ func (state *DockerTaskEngineState) storeV3EndpointIDToDockerIDUnsafe(v3Endpoint
 	state.v3EndpointIDToDockerID[v3EndpointID] = dockerID
 }
 
-// ContainerByV3EndpointID returns an apicontainer.DockerContainer for a given v3 endpoint ID
-func (state *DockerTaskEngineState) ContainerByV3EndpointID(v3EndpointID string) (*apicontainer.DockerContainer, bool) {
+// DockerIDByV3EndpointID returns a docker ID for a given v3 endpoint ID
+func (state *DockerTaskEngineState) DockerIDByV3EndpointID(v3EndpointID string) (string, bool) {
 	state.lock.RLock()
 	defer state.lock.RUnlock()
 
 	dockerID, ok := state.v3EndpointIDToDockerID[v3EndpointID]
-	if !ok {
-		return nil, false
-	}
-
-	container, ok := state.idToContainer[dockerID]
```

Comment:
this looks harmless but this change scares me. Is it ever different?

Reply:
The purpose of this change is to reduce redundant code logic (because only `dockerID` is needed for V3 handler) and not make too many changes to the existing v2 handlers code. The V3 handler package is the only one that uses this method, so it's safe to change this.

This is also explained in the `Implementation details`.